src/read.rs

```diff
@@ -226,48 +225,45 @@ impl<R> Reader<R>
     /// The format is determined by attempting to construct each specific format reader until one
     /// is successful.
     pub fn new(mut reader: R) -> Result<Self, ReadError> {
-        let mut reader_errors = Vec::new();
 
         #[cfg(feature="wav")]
         {
-            let maybe_err = match hound::WavReader::new(&mut reader) {
-                Ok(_) => None,
-                Err(err) => Some(err),
+            let is_wav = match hound::WavReader::new(&mut reader) {
+                Err(hound::Error::FormatError(_)) => false,
+                _ => true,
```

Comment:
@est31 I added a commit that changes the behaviour to return as soon as possible on non-format related errors.